api/clients/v2/coretypes/encoded_payload.go

```diff
@@ -44,6 +44,16 @@ func newEncodedPayload(payload *Payload) (*encodedPayload, error) {
 
 // decode applies the inverse of PayloadEncodingVersion0 to an encodedPayload, and returns the decoded Payload
 func (ep *encodedPayload) decode() (*Payload, error) {
+	if len(ep.bytes) < 32 {
+		return nil, fmt.Errorf("encoded payload must be at least 32 bytes long, but got %d bytes", len(ep.bytes))
+	}
+	if ep.bytes[0] != 0x00 {
+		return nil, fmt.Errorf("encoded payload header first byte must be 0x00, but got %x", ep.bytes[0])
+	}
+	if ep.bytes[1] != byte(codecs.PayloadEncodingVersion0) {
+		return nil, fmt.Errorf("encoded payload header version byte must be %x, but got %x", codecs.PayloadEncodingVersion0, ep.bytes[1])
```

Comment:
@litt3 not sure why this was missing but it feels like a bug right? Or am I missing something?
When adding the new blobDecoding test I realized this wasnt checking the first 2 bytes, which feels wrong.

Reply:
Yeah, I think you're probably right that this should be considered a bug.

I think my thought process was something along the lines of "these bytes are reserved for future use, but aren't relevant right now". But I agree they should be checked.

I'll do a more thorough review of the PR changes to double check, though